core/pkg/server/sender.go

```diff
@@ -441,13 +441,19 @@ func (s *Sender) sendJobFlush() {
 
 	output, err := s.runSummary.CloneTree()
 	if err != nil {
-		s.logger.Error("sender: sendJobFlush: failed to copy run summary", "error", err)
+		s.logger.Error(
+			"sender: sendJobFlush: failed to copy run summary", "error", err,
+		)
 		return
 	}
 
-	artifact, err := s.jobBuilder.Build(output)
+	// If we have a partial job source we just update the
+	// metadata and don't build the job.
```

Comment:
I would move this commend into the job builder. People investigating the sender are unlikely to care about this